### PR TITLE
Add support for ledger_entry RPC

### PR DESCRIPTION
**Pull request by nkramer44**
Adds support for the `ledger_entry` RPC command in XrplClient.

### CODE REVIEW

File: xrpl4j-core/src/main/java/org/xrpl/xrpl4j/model/client/ledger/LedgerEntryResult.java [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+  default boolean validated() {
+    return false;
+  }
+
+  /**
+   * The ID of the ledger entry returned.
+   *
+   * @return The {@link Hash256} representing the ID of the ledger entry.
+   */
+  Hash256 index();
```
Comment (sappenin) [inline xrpl4j-core/src/main/java/org/xrpl/xrpl4j/model/client/ledger/LedgerEntryResult.java:125]: Double-checking -- this is never empty, correct?

File: xrpl4j-core/src/main/java/org/xrpl/xrpl4j/model/ledger/TicketObject.java [hunk 2]
```diff
@@ -110,4 +110,11 @@ default Flags flags() {
    */
   @JsonProperty("TicketSequence")
   UnsignedInteger ticketSequence();
+
+  /**
+   * The unique ID of the {@link TicketObject}.
+   *
+   * @return A {@link Hash256}.
+   */
+  Hash256 index();
```
Comment (sappenin) [inline xrpl4j-core/src/main/java/org/xrpl/xrpl4j/model/ledger/TicketObject.java:119]: Thinking a bit more - should this be pushed up into `LedgerObject`?

File: xrpl4j-core/src/main/java/org/xrpl/xrpl4j/model/client/ledger/LedgerEntryResult.java (hunk 1 above)
Comment (nkramer44) [reply xrpl4j-core/src/main/java/org/xrpl/xrpl4j/model/client/ledger/LedgerEntryResult.java:125]: I don't think so. None of the other `LedgerObject`s have it as `Optional` and I've never seen those break during deserialization.

File: xrpl4j-core/src/main/java/org/xrpl/xrpl4j/model/ledger/TicketObject.java (hunk 2 above)
Comment (nkramer44) [reply xrpl4j-core/src/main/java/org/xrpl/xrpl4j/model/ledger/TicketObject.java:119]: I'm always a bit hesitant to push fields up to a super class in xrpl4j. I _think_ all ledger object responses have this field, but the code in rippled is unclear and who knows if all future ledger objects will have this field in RPC responses. If it turns out one particular `LedgerObject` has an optional `index` field, or it doesn't have an `index` field and we push `index` up to `LedgerObject`, we'd have to break every single `LedgerObject`, whereas if we copy this field in each `LedgerObject` subclass, the damage is limited to just one object. wdyt?

Comment (sappenin) [reply xrpl4j-core/src/main/java/org/xrpl/xrpl4j/model/ledger/TicketObject.java:119]: Yeah, probably the right way to go -- I suppose for now this is just an implementation detail, and does leave room for something different (as you point out), as long as we don't omit this field by accident on some `LedgerObject` that should have it.

**Review by sappenin — APPROVED** [review]

**Pull request merged**